pkg/sql/sem/builtins/geo_builtins.go

```diff
@@ -530,6 +530,47 @@ var geoBuiltins = map[string]builtinDefinition{
 			tree.VolatilityImmutable,
 		),
 	),
+	"st_makepolygon": makeBuiltin(
+		defProps(),
+		geometryOverload1(
+			func(ctx *tree.EvalContext, outer *tree.DGeometry) (tree.Datum, error) {
+				g, err := geomfn.MakePolygon(outer.Geometry)
+				if err != nil {
+					return nil, err
+				}
+				return tree.NewDGeometry(g), nil
+			},
+			types.Geometry,
+			infoBuilder{
+				info: `Returns a new Polygon with the given outer linestring.`,
+			},
+			tree.VolatilityImmutable,
+		),
+		tree.Overload{
+			Types: tree.ArgTypes{
+				{"outer", types.Geometry},
+				{"interior", types.AnyArray},
+			},
+			ReturnType: tree.FixedReturnType(types.Geometry),
+			Fn: func(ctx *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				outer := args[0].(*tree.DGeometry)
+				interiorArr := tree.MustBeDArray(args[1])
+				interior := make([]*geo.Geometry, len(interiorArr.Array))
+				for i, v := range interiorArr.Array {
+					interior[i] = v.(*tree.DGeometry).Geometry
```

Comment:
We should probably check the type assertion holds true lest we panic, i.e. `interiorRing, ok := v.(*tree.DGeometry)`  first. Add a test for this too :)

Reply:
Added a check for this. I also updated the PR description for Postgres and CockroachDB behaviors for `ST_MakePolygon`
Thanks!